src/scene/Scale.js

```diff
@@ -187,8 +193,34 @@ function ordinal(scale, rng, group) {
     };
   }
 
-  prev.range = rng;
-  this._updated = true;
+  // Ordinal scale inversion requires range in ascending order.
+  // Shuffle domain/range to ensure this is true, and cache.
+  if (this._updated) {
```

Comment:
Ah shoot. Your comment on `_updated` points out a control flow flaw here. If we do an early return, the inverse structures will never be recomputed. 

Re: modifying the search procedure, I was hoping to do that initially but haven't been able to think of a way that generalizes for arbitrarily unordered numeric ranges that doesn't amount to just walking the arrays on every invert call. So, given the typical space/time tradeoff, I figured that since scale inversion demands interactive performance, the additional space was worth it. 

Let me know if you think of a better strategy that I've overlooked!

```json
{
  "scales": [
    {
      "name": "s",
      "type": "ordinal",
      "domain": ["a", "b", "c"],
      "range": [50, 3, 25]
    }  
  ]
}
```